dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -151,7 +151,10 @@ def show
       return
     end
 
-    if request.path != (canonical_path = build_script_level_path(@script_level, @extra_params)) && params[:view] != 'summary'
+    course_name = params[:course_course_name]
+    unit_position = params[:unit_position]
+    canonical_path = build_script_level_path(@script_level, @extra_params, course_name: course_name, unit_position: unit_position)
+    if request.path != canonical_path && params[:view] != 'summary'
```

Comment:
I'm having trouble understanding these lines. it looks like passing course_name to build_script_level_path will make /courses/.../units/.../lessons/.../levels/... be the canonical path, and then cause /s/ routes to redirect there. I don't see that happening when I try it out though. can you help me understand the changes to the canonical path on these lines?

Reply:
I see what you're doing now. existing test coverage for `/s/` paths redirecting to canonical `/s/` urls looks a little thin, but I did find these 2 test cases which give me some confidence that you won't be redirecting `/s/` paths to `/courses/`:
* https://github.com/code-dot-org/code-dot-org/blob/3fd01f94c8f42b1e389bf6d63b902ec267e3c796/dashboard/test/controllers/script_levels_controller_test.rb#L922
* https://github.com/code-dot-org/code-dot-org/blob/3fd01f94c8f42b1e389bf6d63b902ec267e3c796/dashboard/test/controllers/script_levels_controller_test.rb#L943

If you are aware of any other cases where we rely on redirects to canonical urls those could be worth adding test cases for... but if nothing else jumps out at you then no worries!